cmd/util/ledger/migrations/broken_contract_migration.go

```diff
@@ -0,0 +1,85 @@
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// BrokenContractMigration fixes some of the early contracts that has been broken due to cadence upgrades.
+func BrokenContractMigration(payloads []ledger.Payload) ([]ledger.Payload, error) {
+	l := NewView(payloads)
+	st := state.NewState(l)
+	sth := state.NewStateHolder(st)
+	a := state.NewAccounts(sth)
+
+	err := migrateContractForAccount(a, "937cbdee135c656c")
+	if err != nil {
+		return nil, err
+	}
+
+	err = migrateContractForAccount(a, "7f560d7e3ff04a31")
+	if err != nil {
+		return nil, err
+	}
+
+	return l.Payloads(), nil
+}
+
+func migrateContractForAccount(accounts *state.Accounts, addressInHex string) error {
+	address := flow.HexToAddress(addressInHex)
+	ok, err := accounts.Exists(address)
+	if err != nil {
+		return err
+	}
+	if ok {
+		err = accounts.SetContract("TokenHolderKeyManager", address, GetKeyManagerContractContent())
+		if err != nil {
+			return err
+		}
+	} else {
+		// if not exist log and return gracefully
+		fmt.Println("warning account does not exist: ", addressInHex)
+	}
+	return nil
+}
+
+func GetKeyManagerContractContent() []byte {
+	return []byte(`import KeyManager from 0x840b99b76051d886
```

Comment:
@psiemens could you please double-check the account address that keyManager is loaded from.

Reply:
Yep, this is correct. 👍